internal/access/signup.go

```diff
@@ -27,10 +27,16 @@ func Signup(c *gin.Context, keyExpiresAt time.Time, baseDomain string, details S
 
 	details.Org.Domain = SanitizedDomain(details.SubDomain, baseDomain)
 
-	if err := data.CreateOrganizationAndSetContext(db, details.Org); err != nil {
+	if err := data.CreateOrganization(db, details.Org); err != nil {
 		return nil, "", fmt.Errorf("create org on sign-up: %w", err)
 	}
 
+	db = data.NewTransaction(db.GormDB(), details.Org.ID)
+	c.Set("db", db)
+	rCtx := GetRequestContext(c)
+	rCtx.DBTxn = db
+	c.Set(RequestContextKey, rCtx)
```

Comment:
Now that we're passing orgID as part of the transaction the `data.CreateOrganization` function can't set the org ID in the context. I think this is a good thing, as previously it was surprising that creating an org mutated the DB to make all queries use that org. 

As a consequence of that change we have to set the tx here for the request of the request, but this is the only place where we need to do this.

I've also restored the original name for `CreateOrganization`, now that it's not doing the `SetContext` part.